### PR TITLE
Updated CMakeLists.txt to accurately name output file under Windows

### PR DESCRIPTION
**Pull request by namreeb**
Various linking options are now also toggle-able with cmake

### CODE REVIEW

**Comment by namreeb** [comment]
Ping

**Comment by ladislav-zezula** [comment]
Wait - the only purpose for CMakeLists.txt, when used with Windows CMAKE, is to generate a project for Visual Studio 2015?

**Comment by namreeb** [comment]
Well, when used with the Visual Studio 2015 generator, yes.  Note that Visual Studio 2017 can also load a CMakeLists.txt file as a 'solution' or .sln file.

File: CMakeLists.txt [hunk 1]
```diff
@@ -288,14 +294,26 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL FreeBSD)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DO_LARGEFILE=0 -Dstat64=stat -Dlstat64=lstat -Dlseek64=lseek -Doff64_t=off_t -Dfstat64=fstat -Dftruncate64=ftruncate")
 endif()
 
-option(BUILD_SHARED_LIBS "Compile shared libraries" OFF)
+option(BUILD_SHARED_LIBS "Link against dependent libraries dynamically" ON)
+option(BUILD_DYNAMIC_MODULE "Output dynamically loadable module" OFF)
```
Comment (Lectem) [inline CMakeLists.txt]: Why did you introduce BUILD_DYNAMIC_MODULE ?
I think you misunderstood how add_library works, calling it with SHARED/STATIC creates a shared or static library. It has *nothing* to do with the MSVC runtime versions (selected using /MD(d) or /MT(d).
By default cmake always use MD(d).
For correctness, your code should not do the following

    set_target_properties(${LIBRARY_NAME} PROPERTIES OUTPUT_NAME_DEBUG StormLibDUS)
    set_target_properties(${LIBRARY_NAME} PROPERTIES OUTPUT_NAME_RELEASE StormLibRUS)

unless you actually pass the /MT(d) flags.
Anyway most of what was done here looks wrong, so I will probably fix it after #113 is merged.

File: CMakeLists.txt [hunk 2]
```diff
@@ -308,10 +326,27 @@ if(BUILD_SHARED_LIBS)
...
-    # On Win32, build StormLib.dll since we don't want to clash with Storm.dll
-    if(WIN32)
-        set_target_properties(${LIBRARY_NAME} PROPERTIES OUTPUT_NAME StormLib)
+if(WIN32)
+    if (UNICODE)
+        if (BUILD_SHARED_LIBS)
+            set_target_properties(${LIBRARY_NAME} PROPERTIES OUTPUT_NAME_DEBUG StormLibDUD)
+            set_target_properties(${LIBRARY_NAME} PROPERTIES OUTPUT_NAME_RELEASE StormLibRUD)
+        else()
+            set_target_properties(${LIBRARY_NAME} PROPERTIES OUTPUT_NAME_DEBUG StormLibDUS)
```
Comment (Lectem) [inline CMakeLists.txt]: See previous comment, this is not correct as cmake will link with /MD by default.

File: CMakeLists.txt [hunk 3]
```diff
@@ -308,10 +326,27 @@ if(BUILD_SHARED_LIBS)
...
     endif()
+else()
+    message(STATUS "Linking against dependent libraries statically")
+endif()
 
-    # On Win32, build StormLib.dll since we don't want to clash with Storm.dll
-    if(WIN32)
-        set_target_properties(${LIBRARY_NAME} PROPERTIES OUTPUT_NAME StormLib)
+if(WIN32)
+    if (UNICODE)
```
Comment (Lectem) [inline CMakeLists.txt]: This is wrong too, unless you add -DUNICODE -D_UNICODE to the list of defines

**Pull request merged**